Update flowjo

hwlans opened the pull request:
**Have you confirmed this pull request is not a duplicate?**
Yes

**Is this pull request creating or modifying a label in the fragments/labels folder, and not Installomator.sh itself?**
Yes

**Did you use [our editorconfig file](https://github.com/Installomator/Installomator/wiki/Contributing-to-Installomator)?**
Yes

**Additional context** Add any other context about the label or fix here.
Updated `downloadURL`

**Installomator log**

Please identify any issues fixed by your pull request by including the issue number. (Example: "Fixes #XXXX")

````
./assemble.sh flowjo
2025-03-31 21:32:00 : INFO  : flowjo : Total items in argumentsArray: 0
2025-03-31 21:32:00 : INFO  : flowjo : argumentsArray:
2025-03-31 21:32:00 : REQ   : flowjo : ################## Start Installomator v. 10.9beta, date 2025-03-31
2025-03-31 21:32:00 : INFO  : flowjo : ################## Version: 10.9beta
2025-03-31 21:32:00 : INFO  : flowjo : ################## Date: 2025-03-31
2025-03-31 21:32:00 : INFO  : flowjo : ################## flowjo
2025-03-31 21:32:00 : DEBUG : flowjo : DEBUG mode 1 enabled.
2025-03-31 21:32:01 : INFO  : flowjo : Reading arguments again:
2025-03-31 21:32:01 : DEBUG : flowjo : name=FlowJo
2025-03-31 21:32:01 : DEBUG : flowjo : appName=
2025-03-31 21:32:01 : DEBUG : flowjo : type=dmg
2025-03-31 21:32:01 : DEBUG : flowjo : archiveName=
2025-03-31 21:32:01 : DEBUG : flowjo : downloadURL=https://fjinstallers.s3.amazonaws.com/FlowJo/FlowJo-OSX64-10.10.0.dmg
2025-03-31 21:32:01 : DEBUG : flowjo : curlOptions=
2025-03-31 21:32:01 : DEBUG : flowjo : appNewVersion=10.10.0
2025-03-31 21:32:01 : DEBUG : flowjo : appCustomVersion function: Not defined
2025-03-31 21:32:01 : DEBUG : flowjo : versionKey=CFBundleShortVersionString
2025-03-31 21:32:01 : DEBUG : flowjo : packageID=
2025-03-31 21:32:01 : DEBUG : flowjo : pkgName=
2025-03-31 21:32:01 : DEBUG : flowjo : choiceChangesXML=
2025-03-31 21:32:01 : DEBUG : flowjo : expectedTeamID=C79HU5AD9V
2025-03-31 21:32:01 : DEBUG : flowjo : blockingProcesses=
2025-03-31 21:32:01 : DEBUG : flowjo : installerTool=
2025-03-31 21:32:01 : DEBUG : flowjo : CLIInstaller=
2025-03-31 21:32:01 : DEBUG : flowjo : CLIArguments=
2025-03-31 21:32:01 : DEBUG : flowjo : updateTool=
2025-03-31 21:32:01 : DEBUG : flowjo : updateToolArguments=
2025-03-31 21:32:01 : DEBUG : flowjo : updateToolRunAsCurrentUser=
2025-03-31 21:32:01 : INFO  : flowjo : BLOCKING_PROCESS_ACTION=tell_user
2025-03-31 21:32:01 : INFO  : flowjo : NOTIFY=success
2025-03-31 21:32:01 : INFO  : flowjo : LOGGING=DEBUG
2025-03-31 21:32:01 : INFO  : flowjo : LOGO=/System/Applications/App Store.app/Contents/Resources/AppIcon.icns
2025-03-31 21:32:01 : INFO  : flowjo : Label type: dmg
2025-03-31 21:32:01 : INFO  : flowjo : archiveName: FlowJo.dmg
2025-03-31 21:32:01 : INFO  : flowjo : no blocking processes defined, using FlowJo as default
2025-03-31 21:32:01 : DEBUG : flowjo : Changing directory to /Users/h.lans/Documents/GitHub/Installomator/build
2025-03-31 21:32:01 : INFO  : flowjo : name: FlowJo, appName: FlowJo.app
2025-03-31 21:32:01 : WARN  : flowjo : No previous app found
2025-03-31 21:32:01 : WARN  : flowjo : could not find FlowJo.app
2025-03-31 21:32:01 : INFO  : flowjo : appversion:
2025-03-31 21:32:01 : INFO  : flowjo : Latest version of FlowJo is 10.10.0
2025-03-31 21:32:01 : REQ   : flowjo : Downloading https://fjinstallers.s3.amazonaws.com/FlowJo/FlowJo-OSX64-10.10.0.dmg to FlowJo.dmg
2025-03-31 21:32:01 : DEBUG : flowjo : No Dialog connection, just download
2025-03-31 21:33:19 : INFO  : flowjo : Downloaded FlowJo.dmg – Type is  zlib compressed data – SHA is 0e75441ba767fb186ca2ff4f4c5fcf9a9a7a453a – Size is 262436 kB
2025-03-31 21:33:19 : DEBUG : flowjo : DEBUG mode 1, not checking for blocking processes
2025-03-31 21:33:19 : REQ   : flowjo : Installing FlowJo
2025-03-31 21:33:19 : INFO  : flowjo : Mounting /Users/h.lans/Documents/GitHub/Installomator/build/FlowJo.dmg
2025-03-31 21:33:22 : DEBUG : flowjo : Debugging enabled, dmgmount output was:
Checksumming Protective Master Boot Record (MBR : 0)…
Protective Master Boot Record (MBR :: verified CRC32 $91F761BF
Checksumming GPT Header (Primary GPT Header : 1)…
GPT Header (Primary GPT Header : 1): verified CRC32 $DF282B6F
Checksumming GPT Partition Data (Primary GPT Table : 2)…
GPT Partition Data (Primary GPT Tabl: verified CRC32 $1880D591
Checksumming  (Apple_Free : 3)…
(Apple_Free : 3): verified CRC32 $00000000
Checksumming disk image (Apple_HFS : 4)…
disk image (Apple_HFS : 4): verified CRC32 $F808F41C
Checksumming  (Apple_Free : 5)…
(Apple_Free : 5): verified CRC32 $00000000
Checksumming GPT Partition Data (Backup GPT Table : 6)…
GPT Partition Data (Backup GPT Table: verified CRC32 $1880D591
Checksumming GPT Header (Backup GPT Header : 7)…
GPT Header (Backup GPT Header : 7): verified CRC32 $7E0A8041
verified CRC32 $821A943F
/dev/disk4          	GUID_partition_scheme
/dev/disk4s1        	Apple_HFS                      	/Volumes/FlowJo Installer

2025-03-31 21:33:22 : INFO  : flowjo : Mounted: /Volumes/FlowJo Installer
2025-03-31 21:33:22 : INFO  : flowjo : Verifying: /Volumes/FlowJo Installer/FlowJo.app
2025-03-31 21:33:22 : DEBUG : flowjo : App size: 445M	/Volumes/FlowJo Installer/FlowJo.app
2025-03-31 21:33:23 : DEBUG : flowjo : Debugging enabled, App Verification output was:
/Volumes/FlowJo Installer/FlowJo.app: accepted
source=Notarized Developer ID
origin=Developer ID Application: FlowJo LLC (C79HU5AD9V)

2025-03-31 21:33:23 : INFO  : flowjo : Team ID matching: C79HU5AD9V (expected: C79HU5AD9V )
2025-03-31 21:33:23 : INFO  : flowjo : Installing FlowJo version 10.10.0 on versionKey CFBundleShortVersionString.
2025-03-31 21:33:23 : DEBUG : flowjo : DEBUG mode 1 enabled, skipping remove, copy and chown steps
2025-03-31 21:33:24 : INFO  : flowjo : Finishing...
2025-03-31 21:33:27 : INFO  : flowjo : name: FlowJo, appName: FlowJo.app
2025-03-31 21:33:27 : WARN  : flowjo : No previous app found
2025-03-31 21:33:27 : WARN  : flowjo : could not find FlowJo.app
2025-03-31 21:33:27 : REQ   : flowjo : Installed FlowJo, version 10.10.0
2025-03-31 21:33:27 : INFO  : flowjo : notifying
2025-03-31 21:33:27 : DEBUG : flowjo : Unmounting /Volumes/FlowJo Installer
2025-03-31 21:33:27 : DEBUG : flowjo : Debugging enabled, Unmounting output was:
"disk4" ejected.
2025-03-31 21:33:27 : DEBUG : flowjo : DEBUG mode 1, not reopening anything
2025-03-31 21:33:27 : REQ   : flowjo : All done!
2025-03-31 21:33:27 : REQ   : flowjo : ################## End Installomator, exit code 0
````

hwlans committed:
Update flowjo.sh

````
./assemble.sh flowjo
2025-03-31 21:32:00 : INFO  : flowjo : Total items in argumentsArray: 0
2025-03-31 21:32:00 : INFO  : flowjo : argumentsArray:
2025-03-31 21:32:00 : REQ   : flowjo : ################## Start Installomator v. 10.9beta, date 2025-03-31
2025-03-31 21:32:00 : INFO  : flowjo : ################## Version: 10.9beta
2025-03-31 21:32:00 : INFO  : flowjo : ################## Date: 2025-03-31
2025-03-31 21:32:00 : INFO  : flowjo : ################## flowjo
2025-03-31 21:32:00 : DEBUG : flowjo : DEBUG mode 1 enabled.
2025-03-31 21:32:01 : INFO  : flowjo : Reading arguments again:
2025-03-31 21:32:01 : DEBUG : flowjo : name=FlowJo
2025-03-31 21:32:01 : DEBUG : flowjo : appName=
2025-03-31 21:32:01 : DEBUG : flowjo : type=dmg
2025-03-31 21:32:01 : DEBUG : flowjo : archiveName=
2025-03-31 21:32:01 : DEBUG : flowjo : downloadURL=https://fjinstallers.s3.amazonaws.com/FlowJo/FlowJo-OSX64-10.10.0.dmg
2025-03-31 21:32:01 : DEBUG : flowjo : curlOptions=
2025-03-31 21:32:01 : DEBUG : flowjo : appNewVersion=10.10.0
2025-03-31 21:32:01 : DEBUG : flowjo : appCustomVersion function: Not defined
2025-03-31 21:32:01 : DEBUG : flowjo : versionKey=CFBundleShortVersionString
2025-03-31 21:32:01 : DEBUG : flowjo : packageID=
2025-03-31 21:32:01 : DEBUG : flowjo : pkgName=
2025-03-31 21:32:01 : DEBUG : flowjo : choiceChangesXML=
2025-03-31 21:32:01 : DEBUG : flowjo : expectedTeamID=C79HU5AD9V
2025-03-31 21:32:01 : DEBUG : flowjo : blockingProcesses=
2025-03-31 21:32:01 : DEBUG : flowjo : installerTool=
2025-03-31 21:32:01 : DEBUG : flowjo : CLIInstaller=
2025-03-31 21:32:01 : DEBUG : flowjo : CLIArguments=
2025-03-31 21:32:01 : DEBUG : flowjo : updateTool=
2025-03-31 21:32:01 : DEBUG : flowjo : updateToolArguments=
2025-03-31 21:32:01 : DEBUG : flowjo : updateToolRunAsCurrentUser=
2025-03-31 21:32:01 : INFO  : flowjo : BLOCKING_PROCESS_ACTION=tell_user
2025-03-31 21:32:01 : INFO  : flowjo : NOTIFY=success
2025-03-31 21:32:01 : INFO  : flowjo : LOGGING=DEBUG
2025-03-31 21:32:01 : INFO  : flowjo : LOGO=/System/Applications/App Store.app/Contents/Resources/AppIcon.icns
2025-03-31 21:32:01 : INFO  : flowjo : Label type: dmg
2025-03-31 21:32:01 : INFO  : flowjo : archiveName: FlowJo.dmg
2025-03-31 21:32:01 : INFO  : flowjo : no blocking processes defined, using FlowJo as default
2025-03-31 21:32:01 : DEBUG : flowjo : Changing directory to /Users/h.lans/Documents/GitHub/Installomator/build
2025-03-31 21:32:01 : INFO  : flowjo : name: FlowJo, appName: FlowJo.app
2025-03-31 21:32:01 : WARN  : flowjo : No previous app found
2025-03-31 21:32:01 : WARN  : flowjo : could not find FlowJo.app
2025-03-31 21:32:01 : INFO  : flowjo : appversion:
2025-03-31 21:32:01 : INFO  : flowjo : Latest version of FlowJo is 10.10.0
2025-03-31 21:32:01 : REQ   : flowjo : Downloading https://fjinstallers.s3.amazonaws.com/FlowJo/FlowJo-OSX64-10.10.0.dmg to FlowJo.dmg
2025-03-31 21:32:01 : DEBUG : flowjo : No Dialog connection, just download
2025-03-31 21:33:19 : INFO  : flowjo : Downloaded FlowJo.dmg – Type is  zlib compressed data – SHA is 0e75441ba767fb186ca2ff4f4c5fcf9a9a7a453a – Size is 262436 kB
2025-03-31 21:33:19 : DEBUG : flowjo : DEBUG mode 1, not checking for blocking processes
2025-03-31 21:33:19 : REQ   : flowjo : Installing FlowJo
2025-03-31 21:33:19 : INFO  : flowjo : Mounting /Users/h.lans/Documents/GitHub/Installomator/build/FlowJo.dmg
2025-03-31 21:33:22 : DEBUG : flowjo : Debugging enabled, dmgmount output was:
Checksumming Protective Master Boot Record (MBR : 0)…
Protective Master Boot Record (MBR :: verified CRC32 $91F761BF
Checksumming GPT Header (Primary GPT Header : 1)…
GPT Header (Primary GPT Header : 1): verified CRC32 $DF282B6F
Checksumming GPT Partition Data (Primary GPT Table : 2)…
GPT Partition Data (Primary GPT Tabl: verified CRC32 $1880D591
Checksumming  (Apple_Free : 3)…
(Apple_Free : 3): verified CRC32 $00000000
Checksumming disk image (Apple_HFS : 4)…
disk image (Apple_HFS : 4): verified CRC32 $F808F41C
Checksumming  (Apple_Free : 5)…
(Apple_Free : 5): verified CRC32 $00000000
Checksumming GPT Partition Data (Backup GPT Table : 6)…
GPT Partition Data (Backup GPT Table: verified CRC32 $1880D591
Checksumming GPT Header (Backup GPT Header : 7)…
GPT Header (Backup GPT Header : 7): verified CRC32 $7E0A8041
verified CRC32 $821A943F
/dev/disk4          	GUID_partition_scheme
/dev/disk4s1        	Apple_HFS                      	/Volumes/FlowJo Installer

2025-03-31 21:33:22 : INFO  : flowjo : Mounted: /Volumes/FlowJo Installer
2025-03-31 21:33:22 : INFO  : flowjo : Verifying: /Volumes/FlowJo Installer/FlowJo.app
2025-03-31 21:33:22 : DEBUG : flowjo : App size: 445M	/Volumes/FlowJo Installer/FlowJo.app
2025-03-31 21:33:23 : DEBUG : flowjo : Debugging enabled, App Verification output was:
/Volumes/FlowJo Installer/FlowJo.app: accepted
source=Notarized Developer ID
origin=Developer ID Application: FlowJo LLC (C79HU5AD9V)

2025-03-31 21:33:23 : INFO  : flowjo : Team ID matching: C79HU5AD9V (expected: C79HU5AD9V )
2025-03-31 21:33:23 : INFO  : flowjo : Installing FlowJo version 10.10.0 on versionKey CFBundleShortVersionString.
2025-03-31 21:33:23 : DEBUG : flowjo : DEBUG mode 1 enabled, skipping remove, copy and chown steps
2025-03-31 21:33:24 : INFO  : flowjo : Finishing...
2025-03-31 21:33:27 : INFO  : flowjo : name: FlowJo, appName: FlowJo.app
2025-03-31 21:33:27 : WARN  : flowjo : No previous app found
2025-03-31 21:33:27 : WARN  : flowjo : could not find FlowJo.app
2025-03-31 21:33:27 : REQ   : flowjo : Installed FlowJo, version 10.10.0
2025-03-31 21:33:27 : INFO  : flowjo : notifying
2025-03-31 21:33:27 : DEBUG : flowjo : Unmounting /Volumes/FlowJo Installer
2025-03-31 21:33:27 : DEBUG : flowjo : Debugging enabled, Unmounting output was:
"disk4" ejected.
2025-03-31 21:33:27 : DEBUG : flowjo : DEBUG mode 1, not reopening anything
2025-03-31 21:33:27 : REQ   : flowjo : All done!
2025-03-31 21:33:27 : REQ   : flowjo : ################## End Installomator, exit code 0
````

anilbagale commented:
2025-04-17 18:24:28 : INFO  : flowjo : Total items in argumentsArray: 1
2025-04-17 18:24:28 : INFO  : flowjo : argumentsArray: INSTALL=force
2025-04-17 18:24:28 : REQ   : flowjo : ################## Start Installomator v. 10.8, date 2025-03-28
2025-04-17 18:24:28 : INFO  : flowjo : ################## Version: 10.8
2025-04-17 18:24:28 : INFO  : flowjo : ################## Date: 2025-03-28
2025-04-17 18:24:28 : INFO  : flowjo : ################## flowjo
2025-04-17 18:24:30 : INFO  : flowjo : Reading arguments again: INSTALL=force
2025-04-17 18:24:30 : ERROR : flowjo : need to provide 'downloadURL'

acodega commented:
So looks like they have Intel and Apple Silicon versions now. The grep needs to be tweaked a little to only capture one URL, and you can reference other labels for how to do choose one URL vs another to determine Intel vs Apple Silicon.

I'm getting this right now with your label:

```
downloadURL=https://downloads.bdaccessportal.com/v11/mac/x64/FlowJo-11.0.0-x64.dmg
https://downloads.bdaccessportal.com/v11/mac/arm64/FlowJo-11.0.0-arm64.dmg
https://downloads.bdaccessportal.com/v11/mac/x64/FlowJo-11.0.0-x64.dmg
https://downloads.bdaccessportal.com/v11/mac/arm64/FlowJo-11.0.0-arm64.dmg
```

acodega commented:
This is ready for testing. 

```
flowjo)
    name="FlowJo"
    type="dmg"
    case $(arch) in
        "arm64")
            cpu_arch="arm64"
            ;;
        "i386")
            cpu_arch="x64"
            ;;
    esac
    downloadPage=$(curl -fs "https://www.flowjo.com/flowjo/download")
    downloadURL=$(echo "$downloadPage" | grep -i -o -E "https.*${cpu_arch}.*\.dmg" | head -n 1)
    appNewVersion=$(echo "$downloadURL" | sed -E 's/.*FlowJo-([0-9.]+)-.*\.dmg/\1/')
    expectedTeamID="C79HU5AD9V"
    ;;
    ```

hwlans committed:
Update flowjo

Flowjo 11 is out and now offers an Apple Silicon version

````
flowjo arm64

./assemble.sh flowjo
2025-06-03 21:22:52 : INFO  : flowjo : Total items in argumentsArray: 0
2025-06-03 21:22:52 : INFO  : flowjo : argumentsArray:
2025-06-03 21:22:52 : REQ   : flowjo : ################## Start Installomator v. 10.9beta, date 2025-06-03
2025-06-03 21:22:52 : INFO  : flowjo : ################## Version: 10.9beta
2025-06-03 21:22:52 : INFO  : flowjo : ################## Date: 2025-06-03
2025-06-03 21:22:52 : INFO  : flowjo : ################## flowjo
2025-06-03 21:22:52 : DEBUG : flowjo : DEBUG mode 1 enabled.
2025-06-03 21:22:54 : INFO  : flowjo : Reading arguments again:
2025-06-03 21:22:54 : DEBUG : flowjo : name=FlowJo 11
2025-06-03 21:22:54 : DEBUG : flowjo : appName=
2025-06-03 21:22:54 : DEBUG : flowjo : type=dmg
2025-06-03 21:22:54 : DEBUG : flowjo : archiveName=
2025-06-03 21:22:54 : DEBUG : flowjo : downloadURL=https://downloads.bdaccessportal.com/v11/mac/arm64/FlowJo-11.0.0-arm64.dmg
2025-06-03 21:22:54 : DEBUG : flowjo : curlOptions=
2025-06-03 21:22:54 : DEBUG : flowjo : appNewVersion=11.0.0
2025-06-03 21:22:54 : DEBUG : flowjo : appCustomVersion function: Not defined
2025-06-03 21:22:54 : DEBUG : flowjo : versionKey=CFBundleShortVersionString
2025-06-03 21:22:54 : DEBUG : flowjo : packageID=
2025-06-03 21:22:54 : DEBUG : flowjo : pkgName=
2025-06-03 21:22:54 : DEBUG : flowjo : choiceChangesXML=
2025-06-03 21:22:54 : DEBUG : flowjo : expectedTeamID=C79HU5AD9V
2025-06-03 21:22:54 : DEBUG : flowjo : blockingProcesses=
2025-06-03 21:22:54 : DEBUG : flowjo : installerTool=
2025-06-03 21:22:54 : DEBUG : flowjo : CLIInstaller=
2025-06-03 21:22:54 : DEBUG : flowjo : CLIArguments=
2025-06-03 21:22:54 : DEBUG : flowjo : updateTool=
2025-06-03 21:22:54 : DEBUG : flowjo : updateToolArguments=
2025-06-03 21:22:54 : DEBUG : flowjo : updateToolRunAsCurrentUser=
2025-06-03 21:22:54 : INFO  : flowjo : BLOCKING_PROCESS_ACTION=tell_user
2025-06-03 21:22:54 : INFO  : flowjo : NOTIFY=success
2025-06-03 21:22:54 : INFO  : flowjo : LOGGING=DEBUG
2025-06-03 21:22:54 : INFO  : flowjo : LOGO=/System/Applications/App Store.app/Contents/Resources/AppIcon.icns
2025-06-03 21:22:54 : INFO  : flowjo : Label type: dmg
2025-06-03 21:22:54 : INFO  : flowjo : archiveName: FlowJo 11.dmg
2025-06-03 21:22:54 : INFO  : flowjo : no blocking processes defined, using FlowJo 11 as default
2025-06-03 21:22:54 : DEBUG : flowjo : Changing directory to /Users/h.lans/Documents/GitHub/Installomator/build
2025-06-03 21:22:54 : INFO  : flowjo : name: FlowJo 11, appName: FlowJo 11.app
2025-06-03 21:22:54 : WARN  : flowjo : No previous app found
2025-06-03 21:22:54 : WARN  : flowjo : could not find FlowJo 11.app
2025-06-03 21:22:54 : INFO  : flowjo : appversion:
2025-06-03 21:22:54 : INFO  : flowjo : Latest version of FlowJo 11 is 11.0.0
2025-06-03 21:22:54 : REQ   : flowjo : Downloading https://downloads.bdaccessportal.com/v11/mac/arm64/FlowJo-11.0.0-arm64.dmg to FlowJo 11.dmg
2025-06-03 21:22:54 : DEBUG : flowjo : No Dialog connection, just download
2025-06-03 21:23:30 : INFO  : flowjo : Downloaded FlowJo 11.dmg – Type is  bzip2 compressed data, block size = 100k – SHA is 137e7fa3afe94e73a8ae5975ebb66ec46e9d7b49 – Size is 360992 kB
2025-06-03 21:23:30 : DEBUG : flowjo : DEBUG mode 1, not checking for blocking processes
2025-06-03 21:23:30 : REQ   : flowjo : Installing FlowJo 11
2025-06-03 21:23:30 : INFO  : flowjo : Mounting /Users/h.lans/Documents/GitHub/Installomator/build/FlowJo 11.dmg
2025-06-03 21:23:50 : DEBUG : flowjo : Debugging enabled, dmgmount output was:
Checksumming Protective Master Boot Record (MBR : 0)…
Protective Master Boot Record (MBR :: verified CRC32 $EBDCBE7B
Checksumming GPT Header (Primary GPT Header : 1)…
GPT Header (Primary GPT Header : 1): verified CRC32 $92825530
Checksumming GPT Partition Data (Primary GPT Table : 2)…
GPT Partition Data (Primary GPT Tabl: verified CRC32 $2EB7EA32
Checksumming  (Apple_Free : 3)…
(Apple_Free : 3): verified CRC32 $00000000
Checksumming disk image (Apple_APFS : 4)…
disk image (Apple_APFS : 4): verified CRC32 $77F2C18B
Checksumming GPT Partition Data (Backup GPT Table : 5)…
GPT Partition Data (Backup GPT Table: verified CRC32 $2EB7EA32
Checksumming GPT Header (Backup GPT Header : 6)…
GPT Header (Backup GPT Header : 6): verified CRC32 $4DD8E659
verified CRC32 $56CC06DB
/dev/disk4          	GUID_partition_scheme
/dev/disk4s1        	Apple_APFS
/dev/disk5          	EF57347C-0000-11AA-AA11-0030654
/dev/disk5s1        	41504653-0000-11AA-AA11-0030654	/Volumes/flowjo-11-11.0.0--arm64

2025-06-03 21:23:50 : INFO  : flowjo : Mounted: /Volumes/flowjo-11-11.0.0--arm64
2025-06-03 21:23:50 : INFO  : flowjo : Verifying: /Volumes/flowjo-11-11.0.0--arm64/FlowJo 11.app
2025-06-03 21:23:50 : DEBUG : flowjo : App size: 593M	/Volumes/flowjo-11-11.0.0--arm64/FlowJo 11.app
2025-06-03 21:24:05 : DEBUG : flowjo : Debugging enabled, App Verification output was:
/Volumes/flowjo-11-11.0.0--arm64/FlowJo 11.app: accepted
source=Notarized Developer ID
origin=Developer ID Application: FlowJo LLC (C79HU5AD9V)

2025-06-03 21:24:05 : INFO  : flowjo : Team ID matching: C79HU5AD9V (expected: C79HU5AD9V )
2025-06-03 21:24:05 : INFO  : flowjo : Installing FlowJo 11 version 11.0.0 on versionKey CFBundleShortVersionString.
2025-06-03 21:24:05 : INFO  : flowjo : App has LSMinimumSystemVersion: 11.0
2025-06-03 21:24:05 : DEBUG : flowjo : DEBUG mode 1 enabled, skipping remove, copy and chown steps
2025-06-03 21:24:05 : INFO  : flowjo : Finishing...
2025-06-03 21:24:08 : INFO  : flowjo : name: FlowJo 11, appName: FlowJo 11.app
2025-06-03 21:24:08 : WARN  : flowjo : No previous app found
2025-06-03 21:24:08 : WARN  : flowjo : could not find FlowJo 11.app
2025-06-03 21:24:08 : REQ   : flowjo : Installed FlowJo 11, version 11.0.0
2025-06-03 21:24:08 : INFO  : flowjo : notifying
2025-06-03 21:24:08 : DEBUG : flowjo : Unmounting /Volumes/flowjo-11-11.0.0--arm64
2025-06-03 21:24:08 : DEBUG : flowjo : Debugging enabled, Unmounting output was:
"disk4" ejected.
2025-06-03 21:24:08 : DEBUG : flowjo : DEBUG mode 1, not reopening anything
2025-06-03 21:24:08 : REQ   : flowjo : All done!
2025-06-03 21:24:08 : REQ   : flowjo : ################## End Installomator, exit code 0

flowjo x64
flowjo x64

./assemble.sh flowjo
2025-06-03 21:19:47 : INFO  : flowjo : Total items in argumentsArray: 0
2025-06-03 21:19:47 : INFO  : flowjo : argumentsArray:
2025-06-03 21:19:47 : REQ   : flowjo : ################## Start Installomator v. 10.9beta, date 2025-06-03
2025-06-03 21:19:47 : INFO  : flowjo : ################## Version: 10.9beta
2025-06-03 21:19:47 : INFO  : flowjo : ################## Date: 2025-06-03
2025-06-03 21:19:47 : INFO  : flowjo : ################## flowjo
2025-06-03 21:19:47 : DEBUG : flowjo : DEBUG mode 1 enabled.
2025-06-03 21:19:48 : INFO  : flowjo : Reading arguments again:
2025-06-03 21:19:48 : DEBUG : flowjo : name=FlowJo 11
2025-06-03 21:19:48 : DEBUG : flowjo : appName=
2025-06-03 21:19:48 : DEBUG : flowjo : type=dmg
2025-06-03 21:19:48 : DEBUG : flowjo : archiveName=
2025-06-03 21:19:48 : DEBUG : flowjo : downloadURL=https://downloads.bdaccessportal.com/v11/mac/x64/FlowJo-11.0.0-x64.dmg
2025-06-03 21:19:48 : DEBUG : flowjo : curlOptions=
2025-06-03 21:19:48 : DEBUG : flowjo : appNewVersion=11.0.0
2025-06-03 21:19:48 : DEBUG : flowjo : appCustomVersion function: Not defined
2025-06-03 21:19:48 : DEBUG : flowjo : versionKey=CFBundleShortVersionString
2025-06-03 21:19:48 : DEBUG : flowjo : packageID=
2025-06-03 21:19:48 : DEBUG : flowjo : pkgName=
2025-06-03 21:19:48 : DEBUG : flowjo : choiceChangesXML=
2025-06-03 21:19:48 : DEBUG : flowjo : expectedTeamID=C79HU5AD9V
2025-06-03 21:19:48 : DEBUG : flowjo : blockingProcesses=
2025-06-03 21:19:48 : DEBUG : flowjo : installerTool=
2025-06-03 21:19:48 : DEBUG : flowjo : CLIInstaller=
2025-06-03 21:19:48 : DEBUG : flowjo : CLIArguments=
2025-06-03 21:19:48 : DEBUG : flowjo : updateTool=
2025-06-03 21:19:48 : DEBUG : flowjo : updateToolArguments=
2025-06-03 21:19:48 : DEBUG : flowjo : updateToolRunAsCurrentUser=
2025-06-03 21:19:48 : INFO  : flowjo : BLOCKING_PROCESS_ACTION=tell_user
2025-06-03 21:19:48 : INFO  : flowjo : NOTIFY=success
2025-06-03 21:19:48 : INFO  : flowjo : LOGGING=DEBUG
2025-06-03 21:19:48 : INFO  : flowjo : LOGO=/System/Applications/App Store.app/Contents/Resources/AppIcon.icns
2025-06-03 21:19:48 : INFO  : flowjo : Label type: dmg
2025-06-03 21:19:48 : INFO  : flowjo : archiveName: FlowJo 11.dmg
2025-06-03 21:19:48 : INFO  : flowjo : no blocking processes defined, using FlowJo 11 as default
2025-06-03 21:19:48 : DEBUG : flowjo : Changing directory to /Users/h.lans/Documents/GitHub/Installomator/build
2025-06-03 21:19:48 : INFO  : flowjo : App(s) found: /Applications/FlowJo 11.app
2025-06-03 21:19:48 : INFO  : flowjo : found app at /Applications/FlowJo 11.app, version 11.0.0, on versionKey CFBundleShortVersionString
2025-06-03 21:19:48 : INFO  : flowjo : appversion: 11.0.0
2025-06-03 21:19:48 : INFO  : flowjo : Latest version of FlowJo 11 is 11.0.0
2025-06-03 21:19:48 : WARN  : flowjo : DEBUG mode 1 enabled, not exiting, but there is no new version of app.
2025-06-03 21:19:48 : REQ   : flowjo : Downloading https://downloads.bdaccessportal.com/v11/mac/x64/FlowJo-11.0.0-x64.dmg to FlowJo 11.dmg
2025-06-03 21:19:48 : DEBUG : flowjo : No Dialog connection, just download
2025-06-03 21:20:25 : INFO  : flowjo : Downloaded FlowJo 11.dmg – Type is  bzip2 compressed data, block size = 100k – SHA is edb8646e7824ae37f055de806ed49718f7e4c151 – Size is 377464 kB
2025-06-03 21:20:25 : DEBUG : flowjo : DEBUG mode 1, not checking for blocking processes
2025-06-03 21:20:25 : REQ   : flowjo : Installing FlowJo 11
2025-06-03 21:20:25 : INFO  : flowjo : Mounting /Users/h.lans/Documents/GitHub/Installomator/build/FlowJo 11.dmg
2025-06-03 21:20:48 : DEBUG : flowjo : Debugging enabled, dmgmount output was:
Checksumming Protective Master Boot Record (MBR : 0)…
Protective Master Boot Record (MBR :: verified CRC32 $5F8AC555
Checksumming GPT Header (Primary GPT Header : 1)…
GPT Header (Primary GPT Header : 1): verified CRC32 $C7B76C6B
Checksumming GPT Partition Data (Primary GPT Table : 2)…
GPT Partition Data (Primary GPT Tabl: verified CRC32 $35F00FF1
Checksumming  (Apple_Free : 3)…
(Apple_Free : 3): verified CRC32 $00000000
Checksumming disk image (Apple_APFS : 4)…
disk image (Apple_APFS : 4): verified CRC32 $06463F64
Checksumming  (Apple_Free : 5)…
(Apple_Free : 5): verified CRC32 $00000000
Checksumming GPT Partition Data (Backup GPT Table : 6)…
GPT Partition Data (Backup GPT Table: verified CRC32 $35F00FF1
Checksumming GPT Header (Backup GPT Header : 7)…
GPT Header (Backup GPT Header : 7): verified CRC32 $7D18ED49
verified CRC32 $B27703ED
/dev/disk4          	GUID_partition_scheme
/dev/disk4s1        	Apple_APFS
/dev/disk5          	EF57347C-0000-11AA-AA11-0030654
/dev/disk5s1        	41504653-0000-11AA-AA11-0030654	/Volumes/flowjo-11-11.0.0-

2025-06-03 21:20:48 : INFO  : flowjo : Mounted: /Volumes/flowjo-11-11.0.0-
2025-06-03 21:20:48 : INFO  : flowjo : Verifying: /Volumes/flowjo-11-11.0.0-/FlowJo 11.app
2025-06-03 21:20:48 : DEBUG : flowjo : App size: 622M	/Volumes/flowjo-11-11.0.0-/FlowJo 11.app
2025-06-03 21:21:04 : DEBUG : flowjo : Debugging enabled, App Verification output was:
/Volumes/flowjo-11-11.0.0-/FlowJo 11.app: accepted
source=Notarized Developer ID
origin=Developer ID Application: FlowJo LLC (C79HU5AD9V)

2025-06-03 21:21:04 : INFO  : flowjo : Team ID matching: C79HU5AD9V (expected: C79HU5AD9V )
2025-06-03 21:21:04 : INFO  : flowjo : Downloaded version of FlowJo 11 is 11.0.0 on versionKey CFBundleShortVersionString, same as installed.
2025-06-03 21:21:04 : DEBUG : flowjo : Unmounting /Volumes/flowjo-11-11.0.0-
2025-06-03 21:21:04 : DEBUG : flowjo : Debugging enabled, Unmounting output was:
"disk4" ejected.
2025-06-03 21:21:04 : DEBUG : flowjo : DEBUG mode 1, not reopening anything
2025-06-03 21:21:04 : REG   : flowjo : No new version to install
2025-06-03 21:21:04 : REQ   : flowjo : ################## End Installomator, exit code 0
````

maccy10 commented:
@acodega 

```
2025-06-05 10:49:37 : REQ   : flowjo : Installing FlowJo
2025-06-05 10:49:37 : INFO  : flowjo : Mounting /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.cfO7SdgZs6/FlowJo.dmg
2025-06-05 10:50:01 : INFO  : flowjo : Mounted: /Volumes/flowjo-11-11.0.0--arm64
2025-06-05 10:50:01 : INFO  : flowjo : Installomator did not close any apps, so no need to reopen any apps.
2025-06-05 10:50:01 : ERROR : flowjo : ERROR: could not find: /Volumes/flowjo-11-11.0.0--arm64/FlowJo.app
2025-06-05 10:50:01 : REQ   : flowjo : ################## End Installomator, exit code 8 

```

jvaglivelo commented:
I was able to get it working by including @acodega changes and adding `appName="FlowJo 11.app"`.

FlowJo versions below 11 had the app named FlowJo.app in the .dmg, and now its named FlowJo 11.app.

Only issue is now I have FlowJo.app and FlowJo 11.app installed.

```
2025-09-05 09:14:47 : INFO  : flowjo : found app at /Applications/FlowJo.app, version 10.10.0, on versionKey CFBundleShortVersionString
2025-09-05 09:14:47 : INFO  : flowjo : appversion: 10.10.0
2025-09-05 09:14:47 : INFO  : flowjo : Latest version of FlowJo is 11.0.2
2025-09-05 09:14:47 : REQ   : flowjo : Downloading https://downloads.bdaccessportal.com/v11/mac/arm64/FlowJo-11.0.2-arm64.dmg to FlowJo.dmg
2025-09-05 09:15:17 : INFO  : flowjo : Downloaded FlowJo.dmg – Type is  bzip2 compressed data, block size = 100k – SHA is ef7e20bad1d910249cefc45f189f8dfe919df3c4 – Size is 360484 kB
2025-09-05 09:15:17 : REQ   : flowjo : no more blocking processes, continue with update
2025-09-05 09:15:17 : REQ   : flowjo : Installing FlowJo
2025-09-05 09:15:17 : INFO  : flowjo : Mounting /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.kCLQxrvVJV/FlowJo.dmg
2025-09-05 09:15:38 : INFO  : flowjo : Mounted: /Volumes/flowjo-11-11.0.2--arm64
2025-09-05 09:15:38 : INFO  : flowjo : Verifying: /Volumes/flowjo-11-11.0.2--arm64/FlowJo 11.app
2025-09-05 09:15:53 : INFO  : flowjo : Team ID matching: C79HU5AD9V (expected: C79HU5AD9V )
2025-09-05 09:15:53 : INFO  : flowjo : Downloaded version of FlowJo is 11.0.2 on versionKey CFBundleShortVersionString (replacing version 10.10.0).
2025-09-05 09:15:53 : INFO  : flowjo : App has LSMinimumSystemVersion: 11.0
2025-09-05 09:15:53 : INFO  : flowjo : Copy /Volumes/flowjo-11-11.0.2--arm64/FlowJo 11.app to /Applications
2025-09-05 09:16:00 : WARN  : flowjo : Changing owner to jvaglivelo
2025-09-05 09:16:00 : INFO  : flowjo : Finishing...
2025-09-05 09:16:03 : INFO  : flowjo : App(s) found: /Applications/FlowJo 11.app
2025-09-05 09:16:03 : INFO  : flowjo : found app at /Applications/FlowJo 11.app, version 11.0.2, on versionKey CFBundleShortVersionString
2025-09-05 09:16:03 : REQ   : flowjo : Installed FlowJo, version 11.0.2
2025-09-05 09:16:03 : INFO  : flowjo : notifying
2025-09-05 09:16:03 : INFO  : flowjo : Installomator did not close any apps, so no need to reopen any apps.
2025-09-05 09:16:03 : REQ   : flowjo : All done!
2025-09-05 09:16:03 : REQ   : flowjo : ################## End Installomator, exit code 0
```

hwlans commented:
There are now two supported versions of FlowJo:
v10 and v11.

FlowJo 10 supports a [license](https://www.flowjo.com/docs/flowjo10/fj-license-setupinfo) via a serialnumber, a usb-dongle or a portal account (username/password)

FlowJo 11 uses a [new license system](https://www.flowjo.com/docs/flowjo11/license-info
): Version 11 is licensed exclusively through the FlowJo Portal licensing system, which employs a username/email + password combination as login credentials

As @jvaglivelo mentionted, FlowJo 11 requires the BD Access Portal.app to be installed from the Flowjo11 DMG file. I'm not sure how we can do that via Installomator?
_"The login/authentication of FlowJo™ v11 is managed by a helper application called BD Access Portal. The BD Access Portal application is included in the installer for FlowJo™ v11 and both applications need to be installed together on the same computer. "_
Source: https://www.flowjo.com/docs/flowjo11/license-info

If you use FlowJo 10 and share the USB-dongle in a labgroup, you'd want to avoid auto upgrading to v11. 

I propose to create a new `flowjo11` label and keep the old `flowjo` label for v10. Thoughts?

uurazzle reviewed (approved):
Hi @hwlans,

Thanks for the updates on this Installomator label. I am approving this PR.

gilburns reviewed (approved):
Thank you for your contribution to the project!
